src/main/java/spiffe/provider/SpiffeProvider.java

```diff
@@ -0,0 +1,35 @@
+package spiffe.provider;
+
+import java.security.Provider;
+import java.security.Security;
+
+import static spiffe.provider.SpiffeProviderConstants.PROVIDER_NAME;
+
+/**
+ * This class represents a Security Provider" for the Java Security API,
+ * It uses a custom implementations of KeyStore and TrustStore Managers that support
+ * Spiffe SVID retrieval and SpiffeID validation
+ *
+ */
+public class SpiffeProvider extends Provider {
+
+    /**
+     * Constructor
+     *
+     * Configure the Provider Name and register KeyManagerFactory, TrustManagerFactory and KeyStore
+     *
+     */
+    public SpiffeProvider() {
+        super(PROVIDER_NAME, 0.1, "");
+        super.put("KeyManagerFactory."+ SpiffeProviderConstants.ALGORITHM, SpiffeKeyManagerFactory.class.getName());
```

Comment:
can you use String.format here?

Reply:
and maybe move to constants, we dont need to recalculate it every time

Reply:
done